### PR TITLE
Feature ingredient class

### PR DESCRIPTION
**Pull request by karmacarlos**
What (if any) features are you implementing?
-Added test for ingredient class

Is there anything that you need from your teammate?
- Please review my test

### CODE REVIEW

**Comment by karmacarlos** [comment]
What (if any) features are you implementing?
-Added code to the implementation files of the Ingredient and Recipe class
-Added code to the test files
What (if anything) did you refactor?
-

Were there any issues that arose?
-I'm having an scope issue with the last test of the Recipe class

Is there anything that you need from your teammate?
- If you can check that last test out would be nice, otherwise I can show you tomorrow.

Any other comments, questions, or concerns?

**Comment by karmacarlos** [comment]
Hi Folks, I fixed the test problem and I'm passing all tests. But that got me another problem with my data for testing (it's too big). But I think thats the only way of achieving a solid test.

File: src/classes/Recipe.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    this.image = recipeDetails.image;
+    this.instructions = recipeDetails.instructions;
+    this.name = recipeDetails.name;
+    this.tags = recipeDetails.tags;
+    this.recipeIngredients = recipeDetails.ingredients;
+    this.processedIngredients = null;
+  }
+
+  processIngredients(dataBase) {
+    this.processedIngredients = this.recipeIngredients.map(ingredientDetails => {
```
Comment (Markus-Xavier) [inline src/classes/Recipe.js]: I wonder if reassigning recipeIngredients would be better here versus making another property? I could see how there could be confusion if we had two properties that are very similar things.

File: src/classes/Recipe.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      let newIngredient = new Ingredient(ingredientDetails);
+      newIngredient.retrieveData(dataBase);
+      newIngredient.calculateCost();
+      return newIngredient;
```
Comment (Markus-Xavier) [inline src/classes/Recipe.js:18]: I like the way the ingredient is being handled here!

**Review by Markus-Xavier — COMMENTED** [review]
I made some comments on specific lines please take a look!

File: src/classes/Recipe.js (hunk 1 above)
Comment (karmacarlos) [reply src/classes/Recipe.js]: I changed this class, we'll review it later

File: src/classes/Recipe.js (hunk 2 above)
Comment (karmacarlos) [reply src/classes/Recipe.js:18]: Thanks Markus.

**Comment by karmacarlos** [comment]
What (if any) features are you implementing?
-Finish Ingredient and recipe class

What (if anything) did you refactor?
-I cleaned up the test files by creating a file with data just for the test

Were there any issues that arose?
-I had to write test for an edge case for when the ingredient class couldn't find the ingredient

File: src/classes/Ingredient.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    this.id = ingredientDetails.id;
+    this.name = this.setName();
+    this.amount = ingredientDetails.quantity.amount;
+    this.unit = ingredientDetails.quantity.unit;
+    this.costInCentsPerUnit = this.setCostInCentsPerUnit();
+    this.ingredientCost = this.calculateCost();
+  }
+
+  setName() {
+    let ingredient = ingredientsDataBase.find(ingredient => (ingredient.id === this.id));
```
Comment (pthan1) [inline src/classes/Ingredient.js:15]: I see that this line and line 24 are identical.  Could there be a way that we can make this code dry?

File: test/Ingredient-test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    assert.equal(ingredient1.amount, 1.5);
+    assert.equal(ingredient2.amount, 0.5);
+  })
+
+  it('should have a unit property', function() {
+    assert.equal(ingredient1.unit, 'c');
+    assert.equal(ingredient2.unit, 'tsp');
+  })
+
+  it('should have a method that retrieves the name from the data manager', function() {
```
Comment (pthan1) [inline test/Ingredient-test.js:47]: It looks like the code is getting the data from an external source and not the data manager.

File: test/Recipe-test.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    assert.equal(recipe1.tags, recipeDetails1.tags);
+    assert.equal(recipe2.tags, recipeDetails2.tags);
+  })
+
+  it('should have a recipe ingredients property', function() {
+    assert.equal(recipe1.recipeIngredients, recipeDetails1.ingredients);
+    assert.equal(recipe2.recipeIngredients, recipeDetails2.ingredients);
+  })
+
+  it('should have a method to process the ingredients', function() {
```
Comment (pthan1) [inline test/Recipe-test.js:51]: I'm having a hard time figuring out what 'process the ingredients' means in this test.  I think the test description and possibly the function names could be more descriptive.

**Review by pthan1 — APPROVED** [review]
